### ISSUE TITLE
Support CLI interface

### ISSUE THREAD

**Issue by alonbhaim** [mockoon/mockoon#62]
Is there way to run mockoon without UI ? I want to use mockoon on UNIX-like operating systems without GUI.

**Comment by 255kb** [comment]
Hello, Thank you for using Mockoon!
There is currently no way to run Mockoon through a command line (terminal, CI env, etc). I think that the amount of work to support this is too high for it to be considered right now. Also, it may be faster to develop something from scratch based on NodeJS / ExpressJS or use an existing tool.

**Comment by kichik** [comment]
It would be useful to be able to configure the mock server using the GUI and then run it automatically for tests in the CLI. Maybe it doesn't have to be Mockoon itself running, but just a minimal server that follows the rules set by the GUI and logs requests to `stdout`.

**Comment by 255kb** [comment]
I am reopening this feature request due to increased interest. However, due to the limited time I have to work on Mockoon this won't be done in a near future.

**Comment by Maurice22** [comment]
Hi @255kb, I wrote a rudimentary CLI wrapper for executing the exported Mockoon environments via the command line [here](https://github.com/Maurice22/mockoon-cli). It ain't pretty but it is doing the job.. Let me know if you don't think this cool, as I mainly used your code and built a wrapper around it...

By doing so, I realized that dummy-json has sever security issues, do you know about that? I needed to fork the repo and fix them myself as the package doe not seem to be actively maintained anymore.

**Comment by 255kb** [comment]
Hi @Maurice22, of course you can, code is published under MIT licence. And it's always cool to see what the community come up with :)
I would just kindly ask you to avoid publishing it under this name on NPM for example. It would complicate things for me when adding this feature in the future. 
Also I would like to take a slightly different path by creating a separated project, extract the code in libraries used by both GUI and CLI, etc which is essential for maintainability.

**Comment by Maurice22** [comment]
Sure thing, I won't publish it as in fact the main reason I put it on GitHub in the first place was to provide some intermediate solution for the people who might need the functionality right now, even if my approach is pretty limited..
I also like your approach better 😃

**Comment by iron66** [comment]
Also would be awesome to release it as npm package.
Imaging, mockoon in devDependencies, `start` script includes mockoon run, configuration and routes are stored in repo and shared with all developers. What an opportunity for development and testing

**Comment by 255kb** [comment]
@iron66 I can imagine :) I already started some necessary refactoring in order to add a CLI. It's definitely on the roadmap now.

**Comment by Liero** [comment]
> a separated project, extract the code in libraries used by both GUI and CLI

You might also consider more lightweight alternative: Just reference the CLI projekt in GUI prokjekt. All the common logic will be in CLI projekt. It does not mean that GUI must invoke CLI interface, you can directly reuse the code from CLI.

Anyway, I'm willing to contribute if you setup [Codespaces ](https://github.com/features/codespaces/) in this repository. It will be super helpful for those willing to contribute

**Comment by 255kb** [comment]
@Liero this is true. But a library shouldn't be that complicated to create, and would maybe be cleaner. 
I am already on it, and after I finish this iteration it will be my main focus

Concerning Codespaces, indeed it would be a good thing, but for now it's in closed beta and I cannot activate it for this organization/repository.
